js/app.js

```diff
@@ -89,39 +127,43 @@ function handleConvertForm(event) {
     .value.trim()
     .toUpperCase();
 
-  // Find the rate object for the from currency
-  const rateObject = currencyRates.find((rate) => rate.base === fromCurrency);
-
-  if (rateObject && rateObject.rates[toCurrency]) {
-    // If the rate object and the exchange rate for the to currency are found
-    const exchangeRate = rateObject.rates[toCurrency]; // Get the exchange rate for the to currency
-    const convertedAmount = amount * exchangeRate; // Calculate the converted amount
+  // Get the value of the amount from the input field
+  const amount = parseFloat(document.getElementById("amount").value);
 
-    // Log the conversion result to the console
-    console.log(
-      `Conversion: ${amount} ${fromCurrency} to ${convertedAmount.toFixed(
-        2
-      )} ${toCurrency}`
-    );
+  // Find the rate
+  const rate = currencyRates.find(
+    (rate) =>
+      rate.baseCurrency === fromCurrency && rate.targetCurrency === toCurrency
+  );
 
-    // Display success message
+  // If rate not found, show an error message
+  if (!rate) {
     displayMessage(
-      `Conversion: ${amount} ${fromCurrency} to ${convertedAmount.toFixed(
-        2
-      )} ${toCurrency}`
+      `Exchange rate from ${fromCurrency} to ${toCurrency} not found.`,
+      false
     );
-  } else {
-    // If no rate is found for the conversion
-    console.log("No rate found for this conversion");
-
-    // Display error message
-    displayMessage("No rate found for this conversion", false);
+    return;
   }
+
+  // Calculate the converted amount
+  const convertedAmount = amount * rate.exchangeRate;
```

Comment:
Your app will need a bit more maths to be more realistic. 
If I know that 1 EUR is 3 USD, how many EUR is 1 USD?